### PR TITLE
Expose `ScriptEditor::get_unsaved_files`

### LINKED ISSUE #13811: Expose `ScriptEditor::get_unsaved_scripts`
### Describe the project you are working on

Patchwork, a real-time version control plugin for Godot.

### Describe the problem or limitation you are having in your project

Currently, there is no way of getting a list of unsaved files in the Script Editor. Therefore, if a plugin wants to write files on-disk, there's no way to know if files are "safe" to update. 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

We simply bind `ScriptEditor::get_unsaved_scripts` to the public API. This allows plugin authors to know both that there **are** unsaved files, as well as **which** files are unsaved and therefore unsafe to update. 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

PR here: https://github.com/godotengine/godot/pull/113762

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I don't know of a (good, stable) workaround that doesn't involve hacking the UI.

### Is there a reason why this should be core and not an add-on in the asset library?

This exposes core functionality, so yes it must be in the core.

### PR DESCRIPTION
**Pull request by LilithSilver**
This PR simply binds `ScriptEditor::get_unsaved_scripts` to GDScript.

Feature description in the proposal: https://github.com/godotengine/godot-proposals/issues/13811

* *Bugsquad edit, closes: https://github.com/godotengine/godot-proposals/issues/13811*

### CODE REVIEW

File: doc/classes/ScriptEditor.xml [hunk 1]
```diff
@@ -48,6 +48,12 @@
 				Returns an array with all [Script] objects which are currently open in editor.
 			</description>
 		</method>
+		<method name="get_unsaved_scripts" qualifiers="const">
+			<return type="PackedStringArray" />
+			<description>
+				Returns an array with the file paths of open and currently unsaved scripts.
```
Comment (AThousandShips) [inline doc/classes/ScriptEditor.xml]:
```suggestion
				Returns an array of file paths of currently unsaved scripts open in the editor.
```
or just, as unsaved naturally means unsaved
```suggestion
				Returns an array of file paths of currently unsaved scripts.
```

File: doc/classes/ScriptEditor.xml [hunk 2]
```diff
@@ -48,6 +48,12 @@
 				Returns an array with all [Script] objects which are currently open in editor.
 			</description>
 		</method>
+		<method name="get_unsaved_scripts" qualifiers="const">
+			<return type="PackedStringArray" />
+			<description>
+				Returns an array of file paths of currently unsaved scripts open in the editor.
```
Comment (Calinou) [inline doc/classes/ScriptEditor.xml]: Use more specific wording (right now, it could be confused for new scripts that haven't been saved yet):

```suggestion
				Returns an array of file paths of scripts with unsaved changes open in the editor.
```

File: doc/classes/ScriptEditor.xml [hunk 3]
```diff
@@ -48,6 +48,12 @@
 				Returns an array with all [Script] objects which are currently open in editor.
 			</description>
 		</method>
+		<method name="get_unsaved_scripts" qualifiers="const">
+			<return type="PackedStringArray" />
+			<description>
+				Returns an array of file paths of scripts with unsaved changes open in the editor.
```
Comment (kitbdev) [inline doc/classes/ScriptEditor.xml:54]: `get_unsaved_scripts` returns a list of names for display, not the file paths. I assume you only want the paths, so a different method is needed.
Also it actually gets all unsaved files, not just scripts.

Comment (nikitalita) [reply doc/classes/ScriptEditor.xml:54]: How would you recommend I implement this? just an additional function that trawls through and gets `get_script_path()` for each script?

Comment (kitbdev) [reply doc/classes/ScriptEditor.xml:54]: It should probably be for all files not just scripts.
A new function can go through all the ScriptEditorBases and the path can be gotten with `seb->get_edited_resource()->get_path()`

Comment (nikitalita) [reply doc/classes/ScriptEditor.xml:54]: Ok, I changed the PR appropriately.

**Review by kitbdev — APPROVED** [review]
Tested, works as expected. Code looks good.
Makes sense to expose this.

Example output including a text file, built-in script, and other scripts: `["res://text_test/test.md", "res://testlive.tscn::GDScript_chd30", "res://text_test/scenes/Test_CodeEdit.gd", "res://text_test/test_script.gd"]`
And it returns an empty array when all are saved.

In certain edge cases, it is possible to get a completely unsaved script open which results in an output like `["", "::4sao5"]`. It's technically correct to return them and should be distinguished from the case when all are saved so its good. Absolute paths may be returned too, if the file is outside of the project. Users may have to account for these depending on the use case.

**Comment by akien-mga** [comment]
Thanks!
